File: quickjs/test/trajtrap.js
import * as a from "liba.so";
function assert(expr) {
    if (!eval(expr)) {
        throw Error(expr);
    }
}
var ctx = new a.trajtrap();
assert("ctx.gen(2, 2, -2, 0, 4) != undefined");
assert("ctx.t != undefined");
assert("ctx.p0 != undefined");
assert("ctx.p1 != undefined");
assert("ctx.v0 != undefined");
assert("ctx.v1 != undefined");
assert("ctx.vc != undefined");
assert("ctx.ta != undefined");
assert("ctx.td != undefined");
assert("ctx.pa != undefined");
assert("ctx.pd != undefined");
assert("ctx.ac != undefined");
assert("ctx.de != undefined");
for (var i = 0; i < ctx.t; i += ctx.t / 10) {
    console.log(i, ctx.pos(i), ctx.vel(i), ctx.acc(i));
}

File: python/test/3rd/trajtrap.py
#!/usr/bin/env python
import os, sys

base = os.path.dirname(__file__)
path = os.path.dirname(base)
path = os.path.dirname(path)
sys.path.insert(0, path)
try:
    import numpy as np
    import matplotlib.pyplot as plt
except Exception as e:
    print(e)
    exit()

import liba  # type: ignore

traj = liba.trajtrap()
traj.gen(2, 2, -2, 0, 4)
data = np.arange(0, traj.t, traj.t / 1000)

plt.figure("trapezoidal velocity trajectory")

plt.subplot(311)
plt.title("trapezoidal velocity trajectory")
plt.ylabel("Position")
plt.plot(data, traj.pos(data), "r-", label="q")
plt.legend()

plt.subplot(312)
plt.ylabel("Velocity")
plt.plot(data, traj.vel(data), "b-", label="v")
plt.legend()

plt.subplot(313)
plt.ylabel("Acceleration")
plt.plot(data, traj.acc(data), "g-", label="a")
plt.legend()

plt.xlabel("t")
plt.savefig(os.path.join(base, "trapezoidal_trajectory.png"))
plt.show()
